components/eamxx/src/control/intensive_observation_period.cpp

```diff
@@ -291,11 +234,14 @@ initialize_iop_file(const util::TimeStamp& run_t0,
   // Initialize time information
   int bdate;
   std::string bdate_name;
-  if      (scorpio::has_variable(iop_file, "bdate"))    bdate_name = "bdate";
-  else if (scorpio::has_variable(iop_file, "basedate")) bdate_name = "basedate";
-  else if (scorpio::has_variable(iop_file, "nbdate"))   bdate_name = "nbdate";
+  if      (scorpio::has_var(iop_file, "bdate"))    bdate_name = "bdate";
+  else if (scorpio::has_var(iop_file, "basedate")) bdate_name = "basedate";
+  else if (scorpio::has_var(iop_file, "nbdate"))   bdate_name = "nbdate";
   else EKAT_ERROR_MSG("Error! No valid name for bdate in "+iop_file+".\n");
-  read_dimensionless_variable_from_file<int>(iop_file, bdate_name, &bdate);
+
+  // Just in case bdate_name is stored with a type that is not "int" (e.g., it's int64)
+  scorpio::change_var_dtype(iop_file,bdate_name,"int");
+  scorpio::read_var(iop_file, bdate_name, &bdate);
```

Comment:
`read_data()` already knows the data type (it's the last arg). Originally, I had `change_var_dtype` called inside `read_var` automatically if the input ptr type did not match the dtype from the nc file. However, for decomposed vars, changing dtype requires to reset the PIO decomposition (since the data type is part of it). Therefore, I decided to force the user to set the var dtype manually.

That said, if you call `change_var_dtype` N times with the same dtype, only the first call does anything, all other calls are no-ops. So maybe I can just call `change_var_dtype` from `read_var` after all...

Reply:
I don't think we'll ever alternate calls to `read_var<T>("file","var",ptr)`, with different data types T, so changing var dtype on the fly is fine.